### PR TITLE
feat(featurecardblockmedium) introducing feature-card-block-medium

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

#3741 

### Description

This adds the `FeatureCardBlockMedium` web component

### Changelog

**New**

- `FeatureCardBlockMedium` web component

**Changed**

- `FeatureCard` core styles moved to mixin

### CODE REVIEW

File: packages/web-components/src/components/feature-card-block-medium/feature-card-block-medium.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @element dds-feature-card-block-medium
+ */
+@customElement(`${ddsPrefix}-feature-card-block-medium`)
+class DDSFeatureCardBlockMedium extends LitElement {
+  static styles = styles;
+
+  protected render() {
+    return html`
+      <div class="${prefix}--feature-card-block-medium__container"><slot name="heading"></slot><slot></slot></div>
```
Comment (asudoh) [inline packages/web-components/src/components/feature-card-block-medium/feature-card-block-medium.ts:29]: Given `<dds-feature-card-block-medium>` itself (host element) and its corresponding style rule (with `:host(dds-feature-card-block-medium)` selector) implements the corresponding element to React `<section class="bx--feature-card-block-medium">`, and `lit-html` doesn't have an error like `Adjacent JSX elements must be wrapped in an enclosing tag`, I think below will work.

```suggestion
      <slot name="heading"></slot><slot></slot>
```

**Review by asudoh — APPROVED** [review]
LGTM 👍, just one thing - Thanks @kennylam!

File: packages/web-components/src/components/feature-card-block-medium/feature-card-block-medium.ts (hunk 1 above)
Comment (kennylam) [reply packages/web-components/src/components/feature-card-block-medium/feature-card-block-medium.ts]: Got it, thanks @asudoh! 👍

Comment (asudoh) [reply packages/web-components/src/components/feature-card-block-medium/feature-card-block-medium.ts]: Cool thanks for the update @kennylam!

**Pull request merged**